---
title: "Simple_rnaseq_analysis"
author: "Mario Ruiz PÃ©rez"
date: "4/6/2022"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Loading data

The data was previously cleaned using unix pipline. The cleansing consisted on removing missing `NA` and zero `0,0` values.

```{r rnaseq}
data = read.csv("genecounts.clean.tsv",sep="\t",header = TRUE, dec=",")

head(data)
```
As it can be seen, data consists of 5 columns. First one being an gene cDNA id and the other 4 the results of the read.

### Length

```{r rnastats}
rows = nrow(data)
```
There is `r rows` columns.

## max cDNA count control_2

```{r}
max(data$control_2)
```

## Scatter plot

```{r}
data$treated_m = rowMeans(data[,c('treated_1','treated_2')])
data$control_m = rowMeans(data[,c('control_1','control_2')])

plot(data$treated_m,data$control_m,
     main="Control vs treated read counts",
     xlab="mean treated counts",
     ylab="mean control counts")

regresion <- lm(treated_m ~ control_m, data=data)
abline(regresion,col='red')
abline(a=0,b=1,col='blue')
```

We can see a little deviation from the expected line, this could mean two different things:

1.The treatment reduces the total rna produced 
2.There is some sort of systematic bias.

The second option is more likely, in either case this should be taken into account when finding differential expressed genes.

Now we use the logarithm of the division to find differential expressed genes. It is used this method instead simple division for making the results more readable, compacting the plot.

```{r}
data$log = log2(data$treated_m/data$control_m)
data$color[data$log > 2.5 | data$log < -2.5] = 'red'
data$color[data$log < 2.5 & data$log > -2.5] = 'grey'

plot(data$log,
     col=data$color,
     main="Differential expression",
     xlab="gene index",
     ylab="log2 treated/controls")
```

It seems to arise a pattern where cDNA counts are slightly under 0 as we expected from the previous plot, we can also observe that there are some measurements which its deviation is 2 to 4 fold, which is up to 20 times more counts. In those cases (2.5 threshold) its more likely that there is a differential expression. More than 4 fold change probably indicates an outlier.



